ospfd/ospf_zebra.c

```diff
@@ -792,6 +786,11 @@ int ospf_redistribute_default_set(struct ospf *ospf, int originate, int mtype,
 			   metric_type(ospf, DEFAULT_ROUTE, 0),
 			   metric_value(ospf, DEFAULT_ROUTE, 0));
 
+	/* Call ospf_is_type_redistributed() second time to verify if
+	 * we have zclient->default_information bitmap set */
+	if (ospf_is_type_redistributed(ospf, DEFAULT_ROUTE, 0))
```

Comment:
This check is redundant, it will always return true since we call `zclient_redistribute_default(ZEBRA_REDISTRIBUTE_DEFAULT_ADD, ...)` a few lines above.

That said, calling `ospf_external_lsa_refresh_default()` fixes the problem as I was expecting. It also fixes #2980. But I think we need to understand why.

@chiragshah6 Please give us a hand. Do you think it makes sense to call `ospf_external_lsa_refresh_default()` here?

Reply:
It would great to know if it’s a bug in the code or why it was added to behave like now 🙂

Reply:
maybe we can skip lsa refresh default inside bitmap checks at all?